content/en/blog/_posts/2022-12-6-provision-volumes-from-cross-namespace-snapshots-alpha.md

```diff
@@ -0,0 +1,133 @@
+date: 2022-12-6
+slug: provision-volumes-from-cross-namespace-snapshots-redesigned
+---
+
+**Authors:**
+Takafumi Takahashi (Hitachi Vantara)
+
+Kubernetes v1.26, released earlier this month, introduced an enable usage of provision of persisten volume claim from volume snapshot in other namespaces.  
+Before Kubernetes v1.25, by using volume snapshots feature, users can provision volumes from snapshots. However, it only works for the `VolumeSnapshot` in the samme namespace, therefore users can't provision a persisten volume claim in one namespace from a `VolumeSnapshot` in other namespace. On the other hand, there are use cases that require to share the `VolumeSnapshot` across namespaces.  
+ To solve this problem,Kubernetes v1.26 includes a new API field called `dataSourceRef2`.
+
+## How it works
+
+Once the csi-provisioner finds a VolumeSnapshot is specified with non-empty namespace as dataSourceRef2, it checks all ReferenceGrants in　PersistentVolumeClaim.spec.dataSourceRef2.namespace to see if access to the snapshot is allowed. If it is allowed, the csi-provisioner provisions a volume from the snapshot.  
+
+## Trying it out
+
+The following things are required to use cross namespace volume provisioning:
+
+* Enable the `CrossNamespaceVolumeDataSource` feature gate
+* Install a CRD for the specific `VolumeSnapShot controller`
+* Install the `VolumeSnapShot controller` itself
+* Install the `External Provisioner controller` itself
+* Install the `Container Stroge Interface Driver` itself
+
+## Putting it all together
+
+To see how this works, you can install the sample and try it out.
+This sample do to create PVC in ns1 namespace from VolumeSnapshot in default namespace by using csi-hostpath driver.
+
+1. Deploy VolumeSnapshot controller.
+
+   ```terminal
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/client/config/crd/snapshot.storage.k8s.io_volumesnapshotclasses.yaml
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/client/config/crd/snapshot.storage.k8s.io_volumesnapshotcontents.yaml
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/client/config/crd/snapshot.storage.k8s.io_volumesnapshots.yaml
+
+
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/deploy/kubernetes/snapshot-controller/rbac-snapshot-controller.yaml
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml
+   ```
+
+2. Deploy External Provisioner controller.
+
+   ```terminal
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-provisioner/master/deploy/kubernetes/deployment.yaml 
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-provisioner/master/deploy/kubernetes/rbac.yaml
+   ```
```

Comment:
I recommend linking to a specific commit, and not to the _master_ branch. That keeps the hyperlink valid.